components/datetime/src/provider/neo.rs

```diff
@@ -0,0 +1,128 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::pattern::{self, runtime};
+use icu_provider::prelude::*;
+
+use zerovec::{ule::UnvalidatedStr, VarZeroVec, ZeroMap};
+
+/// Symbols used for representing the year name
+///
+/// This uses an auxiliary key for length.
```

Comment:
nit: every struct needs the experimental warning

Reply:
yep, i plan to do that once i've got the structure right.